File: 2341. Maximum Number of Pairs in Array/solution.cpp
class Solution {
public:
  vector<int> numberOfPairs(vector<int>& nums) {
    vector<int> result {0, 0};
    unordered_map<int, int> dictionary;
    for (int index = 0; index < nums.size(); index++){
      int num = nums[index];
      dictionary[num]++;
    }

    for (auto num : dictionary){
      result[0] += num.second / 2;
      if (num.second % 2 == 1){
        result[1] += 1;
      }
    }

    return result;
  }
};

File: 2341. Maximum Number of Pairs in Array/solution.py
class Solution(object):
  def numberOfPairs(self, nums):
    """
    :type nums: List[int]
    :rtype: List[int]
    """
    result = [0, 0]
    dictionary = {}
    for num in nums:
      if num not in dictionary:
        dictionary[num] = 1
      else:
        dictionary[num] += 1

    for num in dictionary:
      result[0] += dictionary[num] / 2
      if dictionary[num] % 2 == 1:
        result[1] += 1

    return result